sensorprocess/sensorprocess.go

```diff
@@ -202,44 +250,70 @@ func (config *Config) addIMUReading(
 			AngularVelocity:    tsr.AngularVelocity,
 		}
 
-		// sleep remainder of time interval
-		timeToSleep := tryAddIMUReading(ctx, sr, tsr.ReadingTime, *config)
-		time.Sleep(time.Duration(timeToSleep) * time.Millisecond)
+		// update stored imu time
+		config.Mutex.Lock()
+		config.currentIMUData.time = tsr.ReadingTime
+		config.currentIMUData.data = sr
+		currentLidarData := config.currentLidarData.time
+		cfg := *config
+		config.Mutex.Unlock()
```

Comment:
same comment as above on potentiall moving some of this logic into a helper func

Reply:
Done